File: app/src/main/java/com/test/ashish/sendsmsnew/MessageFragment.java
package com.test.ashish.sendsmsnew;


import android.arch.lifecycle.Observer;
import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.test.ashish.sendsmsnew.adapter.ContactsRecyclerViewAdapter;
import com.test.ashish.sendsmsnew.adapter.MessageRecyclerViewAdapter;
import com.test.ashish.sendsmsnew.pojo.Contact;
import com.test.ashish.sendsmsnew.pojo.Message;

import java.util.ArrayList;


/**
 * A simple {@link Fragment} subclass.
 */
public class MessageFragment extends Fragment {


    private ViewModel viewModel;
    private RecyclerView recyclerView;
    private ArrayList<Message> allMessage;

    public MessageFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view =  inflater.inflate(R.layout.fragment_message, container, false);
        recyclerView = view.findViewById(R.id.recyclerView);
        return view;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        viewModel = (ViewModel) ViewModelProviders.of(this).get(ViewModel.class);

        recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));

        allMessage = new ArrayList<>() ;

        final MessageRecyclerViewAdapter noteAdapter = new MessageRecyclerViewAdapter(allMessage);

        recyclerView.setAdapter(noteAdapter);

        viewModel.getMessageFetchStatus().observe(this, new Observer<Boolean>() {
            @Override
            public void onChanged(@Nullable Boolean aBoolean) {
                if(aBoolean == false){
                    Toast.makeText(getActivity(),"No Match Found", Toast.LENGTH_LONG).show();
                }
            }
        });

        viewModel.getMessages().observe(this,new Observer<ArrayList<Message>>() {
            @Override
            public void onChanged(@Nullable ArrayList<Message> contacts) {
                if(contacts.size()>0) {
                    noteAdapter.setData(contacts);
                }
            }
        });
    }

}

File: app/src/main/java/com/test/ashish/sendsmsnew/MessageFragment.kt
package com.test.ashish.sendsmsnew


import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast

import com.test.ashish.sendsmsnew.adapter.ContactsRecyclerViewAdapter
import com.test.ashish.sendsmsnew.adapter.MessageRecyclerViewAdapter
import com.test.ashish.sendsmsnew.pojo.Contact
import com.test.ashish.sendsmsnew.pojo.Message

import java.util.ArrayList


/**
 * A simple [Fragment] subclass.
 */
class MessageFragment : Fragment() {


    private var viewModel: ViewModel? = null
    private var recyclerView: RecyclerView? = null
    private var allMessage: ArrayList<Message>? = null


    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        val view = inflater.inflate(R.layout.fragment_message, container, false)
        recyclerView = view.findViewById(R.id.recyclerView)
        return view
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        viewModel = ViewModelProviders.of(this).get(ViewModel::class.java)

        recyclerView!!.layoutManager = LinearLayoutManager(activity)

        allMessage = ArrayList()

        val noteAdapter = MessageRecyclerViewAdapter(allMessage!!)

        recyclerView!!.adapter = noteAdapter

        viewModel!!.messageFetchStatus.observe(this, Observer { aBoolean ->
            if (aBoolean == false) {
                Toast.makeText(activity, "No Match Found", Toast.LENGTH_LONG).show()
            }
        })

        viewModel!!.messages.observe(this, Observer { contacts ->
            if (contacts!!.size > 0) {
                noteAdapter.setData(contacts)
            }
        })
    }

}// Required empty public constructor
